npm/metrics/http.go

```diff
@@ -0,0 +1,90 @@
+package metrics
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var started = false
+
+const httpPort = ":8000"
+
+var handler http.Handler
+
+// StartHTTP starts a HTTP endpoint on port 8000. Metrics are exposed on the endpoint /metrics.
+// Set asGoRoutine to true if you want to be able to effectively run other code after calling this.
+func StartHTTP(asGoRoutine bool) {
+	if started {
+		return
+	}
+	started = true
+
+	http.Handle("/metrics", getHandler())
+	if asGoRoutine {
+		go http.ListenAndServe(httpPort, nil)
+	} else {
+		http.ListenAndServe(httpPort, nil)
+	}
+}
+
+// getHandler returns the HTTP handler for the metrics endpoint
+func getHandler() http.Handler {
+	if handler == nil {
+		handler = promhttp.Handler()
+		// 	handler = promhttp.HandlerFor(networkingRegistry, promhttp.HandlerOpts{}) // promhttp.Handler()
```

Comment:
Do we need this commented out code?

Reply:
just an alternative option, but not necessary. I'll delete it